presto-hive/src/main/java/io/prestosql/plugin/hive/HiveMetadata.java

```diff
@@ -1250,7 +1251,20 @@ public HiveOutputTableHandle beginCreateTable(ConnectorSession session, Connecto
                 .collect(toList());
         checkPartitionTypesSupported(partitionColumns);
 
-        LocationHandle locationHandle = locationService.forNewTable(metastore, session, schemaName, tableName);
+        LocationHandle locationHandle;
+        String externalLocation = getExternalLocation(tableMetadata.getProperties());
+        boolean isExternal;
+        if (externalLocation != null) {
+            isExternal = true;
+            // Do not use temporary directory for external tables
+            Path targetPath = getExternalPath(new HdfsContext(session, schemaName, tableName), externalLocation, true);
+            locationHandle = new LocationHandle(targetPath, targetPath, false, DIRECT_TO_TARGET_NEW_DIRECTORY);
```

Comment:
Why cannot we call `locationService.forNewTable(metastore, session, schemaName, tableName)`?